Core/src/main.cpp

```diff
@@ -103,11 +102,10 @@ int run_glfw() {
 	game->Initialize();
 	audioManager->Initialize();
 	
-    // TODO: componentize all of this, so it isn't hard-coded to two sounds/channels
-	FMOD::Sound* backgroundMusic = audioManager->LoadAudio("Assets/Audio/car-motor.mp3");
-    FMOD::Sound* audienceSound = audioManager->LoadAudio("Assets/Audio/audience.mp3");
-	audioManager->PlayDynamicSound(backgroundMusic, true, {0, 50.0f, 0});
-    audioManager->PlayStaticSound(audienceSound, true, { 0, 0.0f, 0 });
+	FMOD::Sound* carMotor = audioManager->LoadAudio("Assets/Audio/car-motor.mp3");
+    AudioComponent* carAudio = new AudioComponent(testCarEntity, audioManager);
+    testCarEntity->addComponent(*carAudio); // TODO: Use in Game, not main, I suppose.
+    carAudio->PlaySound(carMotor, true, false);
```

Comment:
This code block should not be in main it should most likely be in scene. Audio has already done the work of systemitizing audio in an ECS way only to violate it here.

Reply:
Inconsistent Tabs

Reply:
The tabbing is a display error, it seems. The tab existed in the dev branch before my edit, as I never actually changed this line, just removed the other audio and changed the variable name. Tabs and 4 spaces appear identical on my machine as it is configured that way. It has been replaced with 4 spaces. Github seems to be displaying tabs as 8 spaces.

The code block was a modification of existing audio code in main, originally using the non-component implementation.

To move it to a scene, I would need some guidance from someone working on the scene construction. Will do, with help.

Reply:
>To move it to a scene, I would need some guidance from someone working on the scene construction. Will do, with help.

Check Scene.cpp, void Scene::SpawnSceneDefinition()'s method body is where you should be creating components and entities. You can see everyone else's code there as examples to emulate. Such as:

```cpp
    auto garageRoomEnt = entityManager->findFirstEntityByDisplayName("Garage Controller");
    ScriptComponent* garageScriptComponent = new ScriptComponent(garageRoomEnt);
    GarageScriptParams garageParams;
    garageParams.orbitScript = orbScript;
    garageParams.cars.push_back(entityManager->findFirstEntityByDisplayName("Test Car"));
    garageParams.cars.push_back(entityManager->findFirstEntityByDisplayName("TestCar2"));
    garageParams.chosenTarget = garageParams.cars[0]->transform;

    garageScriptComponent->CreateAndAddScript<GarageScript>(&garageParams);
    garageRoomEnt->addComponent(*garageScriptComponent);
```

Reply:
Tabbing: Senseis in the industry and programming books stress the importance that it's not important who caused the issue rather than the fact that there is an issue. This is an opportunity to fix the spacing on main.cpp which appears inconsistent accross the board. This is called code smells and we should address it now that we have a chance.

Scene: I think you should attempt it before pulling someone aside to ask for their help. Imagine a senior dev tasks you with something and then you say i'll do it, but only with someones help. Try it first and then ping the channels if you still need help. You can follow the scene logic from main and go from there. Jas points out SpawnSceneDefinition above which should give you everything you need to know.